addons/web/static/src/views/fields/statusbar_time/statusbar_time_field.js

```diff
@@ -0,0 +1,100 @@
+        this.state.showTimeState = !this.state.showTimeState;
+        browser.localStorage.setItem('showTimeState', this.state.showTimeState);
+    }
+
+    humanizeDuration(durationInSeconds) {
+        const minuteInSeconds = 60;
+        const hourInSeconds = 60 * minuteInSeconds;
+        const dayInSeconds = 24 * hourInSeconds;
+        const weekInSeconds = 7 * dayInSeconds;
+        const monthInSeconds = 30 * dayInSeconds;
+        const yearInSeconds = 365 * dayInSeconds;
+
+        if (durationInSeconds >= yearInSeconds) {
+            const years = Math.floor(durationInSeconds / yearInSeconds);
+            return years + ' ' + (years === 1 ? _t('year') : _t('years'));
+        } else if (durationInSeconds >= monthInSeconds) {
+            const months = Math.floor(durationInSeconds / monthInSeconds);
+            return months + ' ' + (months === 1 ? _t('month') : _t('months'));
+        } else if (durationInSeconds >= weekInSeconds) {
+            const weeks = Math.floor(durationInSeconds / weekInSeconds);
+            return weeks + ' ' + (weeks === 1 ? _t('week') : _t('weeks'));
+        } else if (durationInSeconds >= dayInSeconds) {
+            const days = Math.floor(durationInSeconds / dayInSeconds);
+            return days + ' ' + (days === 1 ? _t('day') : _t('days'));
+        } else if (durationInSeconds >= hourInSeconds) {
+            const hours = Math.floor(durationInSeconds / hourInSeconds);
+            return hours + ' ' + (hours === 1 ? _t('hour') : _t('hours'));
+        }
+    }
+
+    setStatusbarTime(items) {
+        const jsonStatustime = this.props.record.data.json_statustime;
+        if (!Object.keys(jsonStatustime).length) {
+            return;
+        }
+        for (const item of items) {
+            if (item.id in jsonStatustime && jsonStatustime[item.id] >= 3600) {
+                item.timeInStage = this.humanizeDuration(jsonStatustime[item.id]);
+            } else {
+                item.timeInStage = false;
+            }
+        }
+    }
+
+    computeItems(grouped = true) {
+        let items = super.computeItems();
+        if (!grouped) {
+            return items;
+        }
+        if (this.state.showTimeState && this.props.record.data.json_statustime) {
+            for (const property in items) {
+                this.setStatusbarTime(items[property]);
+            }
+        }
+        items.unfolded.forEach((item, index) => {
+            item.isFirstButton = (index === 0);
+        });
+        return items;
+    }
+}
+
+export const statusBarTimeField = {
+    component: StatusBarTimeField,
+    displayName: _t("Status with time"),
+    supportedTypes: ["many2one"],
+    isEmpty: statusBarField.isEmpty,
+    legacySpecialData: statusBarField.legacySpecialData,
+    fieldDependencies: [{ name: "json_statustime", type: "JSON" }],
+    extractProps: ({ attrs, options, viewType }) => ({
+        ...statusBarField.extractProps({ attrs, options, viewType }),
+    }),
+};
```

Comment:
```suggestion
export const statusBarTimeField = {
    ...statusBarField,
    component: StatusBarTimeField,
    displayName: _t("Status with time"),
    supportedTypes: ["many2one"],
    fieldDependencies: [{ name: "json_statustime", type: "JSON" }],
};
```

Reply:
json_statustime field name hardcoded? looks like a bad idea

Reply:
Hello @aab-odoo , thanks for having a look and reviewing - I applied your suggestions

As for the hardcoded field: the widget is made to be used with the new mixin, so a record that uses this widget always has the field - there is no usecase for now where the data would come from another field.
If this changed in the future and other uses were developed for the widget, I feel like that could be changed then